test/falco_tests.yaml

```diff
@@ -20,22 +20,55 @@ trace_files: !mux
   builtin_rules_no_warnings:
     detect: False
     trace_file: trace_files/empty.scap
-    rules_warning: False
 
+  # The rules_events part of this test was mistakenly disabled as a
+  # part of the lua to c++ rules switch. The rule loading
+  # implementation no longer prints messages of the form:
+  # "Event types for rule (<RULE>): (<EVENT TYPES>)
+  # And without that output, none of the checks below rules_events
+  # are considered.
+  # XXX/mstemm add it back
   test_warnings:
     detect: False
     trace_file: trace_files/empty.scap
-    rules_file: rules/falco_rules_warnings.yaml
-    rules_warning:
-      - no_evttype
-      - evttype_not_equals
-      - leading_not
-      - not_equals_at_end
-      - not_at_end
-      - not_equals_and_not
-      - leading_in_not_equals_at_evttype
-      - not_with_evttypes
-      - not_with_evttypes_addl
+    validate_rules_file: rules/falco_rules_warnings.yaml
```

Comment:
Are these all of type `LOAD_NO_EVTTYPE`? Or will we need to change them once we bring back the other kinds of warnings we had in `sinsp_rule_utils.lua`?

Reply:
Yeah, they all ultimately result in the same (single) warning LOAD_NO_EVTTYPE, but have different ways to write a condition that ultimately matches too many event types e.g. "evt.type!=execve", "not evt.type=execve", etc. There's a lot of overlap between these cases and the ones in test_filter_evttype_resolver.cpp. These tests are much older, before we even had unit tests, but I think they're still valuable to test the warning result paths.